src/otx/core/data/dataset/classification.py

```diff
@@ -89,3 +93,171 @@ def _convert_to_onehot(self, labels: torch.tensor) -> torch.tensor:
+
+        i.e.
+        Let's assume that we used the same dataset with example of the definition of HLabelInfo
+        and the original labels are ["Rigid", "Panda", "Lion"].
+
+        Then, h-label format will be [1, -1, 0, 1, 1].
+        The first N-th indices represent the label index of multiclass heads (N=num_multiclass_heads),
+        others represent the multilabel labels.
+
+        [Multiclass Heads: [1, -1]]
+        0-th index = 1 -> ["Non-Rigid"(X), "Rigid"(O)] <- First multiclass head
+        1-st index = -1 -> ["Rectangle"(X), "Triangle"(X)] <- Second multiclass head
+
+        [Multilabel Head: [0, 1, 1]]
+        2, 3, 4 indices = [0, 1, 1] -> ["Circle"(X), "Lion"(O), "Panda"(O)]
+        """
+        num_multiclass_heads = self.hlabel_info.num_multiclass_heads
+        num_multilabel_classes = self.hlabel_info.num_multilabel_classes
+
+        # NOTE: currently ignored labels are not considered yet.
+        ignored_labels: list = []
+
+        class_indices = [0] * (num_multiclass_heads + num_multilabel_classes)
+        for i in range(num_multiclass_heads):
+            class_indices[i] = -1
+
+        for ann in label_anns:
+            ann_name = self.dm_categories.items[ann.label].name
+            group_idx, in_group_idx = self.hlabel_info.class_to_group_idx[ann_name]
+
+            if group_idx < num_multiclass_heads:
+                class_indices[group_idx] = in_group_idx
+            elif ann.label not in ignored_labels:
+                class_indices[num_multiclass_heads + in_group_idx] = 1
+            else:
+                class_indices[num_multiclass_heads + in_group_idx] = -1
+
+        return class_indices
+
+    def _get_hlabel_info(self) -> HLabelInfo:
+        """Get H-label information.
+
+        To check the detailed information, please see the definition of HLabelInfo.
+        """
+
+        def get_exclusive_group_info(all_groups: list[Label | list[Label]]) -> dict[str, Any]:
+            """Get exclusive group information."""
+            exclusive_groups = [g for g in all_groups if len(g) > 1]
+            exclusive_groups.sort(key=itemgetter(0))
+
+            last_logits_pos = 0
+            num_single_label_classes = 0
+            head_idx_to_logits_range = {}
+            class_to_idx = {}
+
+            for i, group in enumerate(exclusive_groups):
+                head_idx_to_logits_range[str(i)] = (last_logits_pos, last_logits_pos + len(group))
+                last_logits_pos += len(group)
+                for j, c in enumerate(group):
+                    class_to_idx[c] = (i, j)
+                    num_single_label_classes += 1
+
+            return {
+                "num_multiclass_heads": len(exclusive_groups),
+                "exclusive_groups": exclusive_groups,
+                "head_idx_to_logits_range": head_idx_to_logits_range,
+                "class_to_idx": class_to_idx,
+                "num_single_label_classes": num_single_label_classes,
+            }
```

Comment:
It is cumbersome but can we stick to static type approaches if possible rather than Python dictionary such as dataclass or namedtuple?